### DISCUSSION TITLE
How to register 2 different static file serve route manually?

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2832] [Q&A] [answered]
```go
func defaultTest1Dir() string {
	if strings.HasPrefix(os.Args[0], os.TempDir()) {
		return "./pb_public/test1"
	}
	return filepath.Join(os.Args[0], "../pb_public/test1")
}

func defaultTest2Dir() string {
	if strings.HasPrefix(os.Args[0], os.TempDir()) {
		return "./pb_public/test2"
	}
	return filepath.Join(os.Args[0], "../pb_public/test2")
}
```
```go
var test1Dir string
database.App.RootCmd.PersistentFlags().StringVar(
	&test1Dir,
	"test1Dir",
	defaultTest1Dir(),
	"the directory to serve static files",
)

var test2Dir string
database.App.RootCmd.PersistentFlags().StringVar(
	&test2Dir,
	"test2Dir",
	defaultTest2Dir(),
	"the directory to serve static files",
)
```

```go
database.App.OnBeforeServe().Add(func(e *core.ServeEvent) error {
	e.Router.GET("/test1/*", apis.StaticDirectoryHandler(os.DirFS(test1Dir), false))
	e.Router.GET("/test2/*", apis.StaticDirectoryHandler(os.DirFS(test2Dir), false))

	return nil
})
```

When I call `/test1/` url its not working like this
![Screenshot 2023-07-02 at 16 43 27](https://github.com/pocketbase/pocketbase/assets/44497354/d62ec508-38c7-46b4-8ac6-42fcc25d4f07)

```go
e.Router.GET("/test1/*", apis.StaticDirectoryHandler(os.DirFS(test1Dir), false))
```
When I change test1 to below its works
```go
e.Router.GET("/*", apis.StaticDirectoryHandler(os.DirFS(test1Dir), false))
```

**Comment by ganigeorgiev** [comment]
It is working fine for me - I've created 2 directories - `./pb_public/test1/` and `./pb_public/test2/` and inside each single `index.html` file.

Make sure that the paths to the directories are correct and accessible (your current working directory can also affect the defaultDir methods). 

Note that the above routes also will require the static files url to start with `/test1/` or `/test2/` (including the trailing slash).

**Reply by Oggy4th** [reply]
Eventhough url is seperated to `test1` and `test2` and dir is seperate to `test1` and `test2` when I try to load it, it tries to retrieve `flutter.js` from root url and not from `test1/flutter.js`

**Reply by ganigeorgiev** [reply]
I'm not sure how to reproduce it.
From where and how `flutter.js` is requested (aka. is the complete url to the `flutter.js` file `http://localhost:8090/test1/flutter.js`)?
Could you provide a minimal reproducible code sample (it is not necessary to include the specific flutter assets)?

**Reply by Oggy4th** [reply]
`/pb_public/test1` and `/pb_public/test2` same like this
![image](https://github.com/pocketbase/pocketbase/assets/44497354/99d2365e-694e-4053-aaea-0548acfaab1d)

**Reply by ganigeorgiev** [reply]
But how `flutter.js` is actually requested?
Is it loaded from inside the `index.html` and if Yes does it use an absolute or relative url?
What is the result of directly accessing the flutter.js file, aka. `http://localhost:8090/test1/flutter.js`?

**Reply by ganigeorgiev** [reply]
Note that in the above error screenshot seems that you are requesting `https://yourdomain.com/flutter.js`, not `https://yourdomain.com/test1/flutter.js`. Make sure that you are using the correct relative or absolute path pointing to the flutter.js file.

**Reply by Oggy4th** [reply]
![image](https://github.com/pocketbase/pocketbase/assets/44497354/05093420-efe6-4543-a860-4f78a2dc59d2)

**Reply by ganigeorgiev** [reply, marked as answer]
@Oggy4th Since you are able to directly access the file, then you should fix the path to the file from insde your index.html (or from anywhere else that requests it; see the comment above https://github.com/pocketbase/pocketbase/discussions/2832#discussioncomment-6335945).

**Reply by Oggy4th** [reply]
Resolved by changing 
```html 
<base href="/test1/">
```
 in `index.html` thank you @ganigeorgiev
